strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-pypi-layout-provider/src/main/java/org/carlspring/strongbox/dependency/snippet/PypiWheelDependencyFormatter.java

```diff
@@ -0,0 +1,65 @@
+package org.carlspring.strongbox.dependency.snippet;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+import org.carlspring.strongbox.artifact.coordinates.PypiWheelArtifactCoordinates;
+import org.carlspring.strongbox.providers.layout.AbstractLayoutProvider;
+import org.carlspring.strongbox.providers.layout.PypiLayoutProvider;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+* This class is an implementation of DependencySynonymFormatter for Pypi artifacts
+*
+* @author whalenda
+**/
+@Component
+public class PypiWheelDependencyFormatter
+  implements DependencySynonymFormatter
+{
+  private static final Logger logger = LoggerFactory.getLogger(AbstractLayoutProvider.class);
+
+  @Inject
+  private CompatibleDependencyFormatRegistry compatibleDependencyFormatRegistry;
+  @PostConstruct
+  @Override
+  public void register()
+  {
+     compatibleDependencyFormatRegistry.addProviderImplementation(getLayout(), getFormatAlias(), this);
+     logger.debug("Initialized the Pypi dependency formatter.");
+  }
+
+  @Override
+  public String getLayout()
+  {
+    return PypiLayoutProvider.ALIAS;
+  }
+
+  @Override
+  public String getFormatAlias()
+  {
+    return PypiLayoutProvider.ALIAS;
+  }
+
+  /**
+  * This method takes in a set of Pypi Artifact Coordinates and returns the properly formatted dependency snippet
+  * @param a PyPiWheelArtifactCoordinates object
+  * @return a string representing a properly formatted dependency snippet
+  **/
+  @Override
+  public String getDependencySnippet(ArtifactCoordinates input_coordinates)
+  {
+    PypiWheelArtifactCoordinates coordinates = (PypiWheelArtifactCoordinates) input_coordinates;
+    String sb = coordinates.getId(); 
+    
+    if (coordinates.getVersion() != "version")
```

Comment:
In Java, strings (and objects in general) should not be compared using `==`. You should use `equals(...)` instead.

For more details, please have a look at these links:
* https://www.leepoint.net/data/expressions/22compareobjects.html
* https://stackoverflow.com/a/13387787/774183
* https://www.geeksforgeeks.org/equals-hashcode-methods-java/

Reply:
In addition, it would be better to change this to: `"version".equals(coordinates.getVersion())`. The reason being, that you know that `"version"` is a not null value, but `coordinates.getVersion` might not be properly initialized and return `null`, at which point you'll hit a `NullPointerException`.